frontend/packages/console-dynamic-plugin-sdk/src/extensions/notification-alert.ts

```diff
@@ -0,0 +1,16 @@
+import { Extension } from '@console/plugin-sdk/src/typings/base';
+import { ExtensionDeclaration, CodeRef } from '../types';
+
+export type AlertAction = ExtensionDeclaration<
+  'console.alert-action',
+  {
+    /* Alert name as defined by `alert.rule.name` property */
+    alert: string;
+    /* Action text */
+    text: string;
+    /* function that returns the path or performs side effects */
+    actionProperty: CodeRef<(alert) => () => void>;
```

Comment:
The action should immediately execute and not be a factory function. If the implementation wants to wrap the action in a factory function, then it can do so instead of forcing the API to be this way.
```suggestion
    action: CodeRef<(alert) => void>;
```